Formula/sdkman-cli.rb

```diff
@@ -52,11 +52,21 @@ def install
     (libexec/"var/version").write version
     (libexec/"var/version_native").write resource("sdkman_cli_native").version
 
-    on_intel do
-      (libexec/"var/platform").write "darwinx64"
+    on_macos do
+      on_arm do
+        (libexec/"var/platform").write "darwinarm64"
+      end
+      on_intel do
+        (libexec/"var/platform").write "darwinx64"
+      end
     end
-    on_arm do
-      (libexec/"var/platform").write "darwinarm64"
+    on_linux do
+      on_arm do
+        (libexec/"var/platform").write "linuxarm64"
+      end
+      on_intel do
+        (libexec/"var/platform").write "linuxx64"
+      end
```

Comment:
@helpermethod 
I get the platform from https://github.com/sdkman/sdkman-cli/blob/89199f5ecd49cf0dc0308e39be085aeff3dd8765/src/test/groovy/sdkman/support/UnixUtils.groovy#L7

After this fix, `SDKMAN_PLATFORM` should be set correctly and `sdk install java 21.0.4-tem` should install the `java` for correct platform.